First time repository setup (#123)
codesandbox/docs: packages/projects-docs/pages/learn/getting-started/setting-up-repository.md

---
title: Repositories Setup + Tips & Tricks
description: This guide is the first step to get started with CodeSandbox for your repo. It’s a walk-through with a bunch of tips and tricks on how to get the most out of CodeSandbox!
---

import { Callout } from 'nextra-theme-docs'

> 

Reviewer: For SEO, we'd need a title here.

Author: Good one!

This guide will help you get started with using CodeSandbox for your repo. It’s a walk-through with a bunch of tips and tricks on how to get the most out of CodeSandbox!

> ## Create a Team & Import Repository

Reviewer: Should we call this section: Import Repository into Team account? It seems like the following paragraph is specific to teams.

Author: Oh that's a good point... It's really optional, I've updated the content a bit to say that we recommend it, but it's not required.

To get started with CodeSandbox for your repo, you first need to create a team. You can do this at [the dashboard](https://codesandbox.io/dashboard), by clicking on the workspace on the left, and pressing “Create a new team”. You can optionally start a free trial, in case you want better VM specs and import a private repositories.

![Pasted image 20230202132514.png](/team-creation.png)

Once you’ve created the team, you can import a repository by clicking “Import from GitHub” and entering the URL of the GitHub repo. Once you’ve clicked “Import”, you’re off to the races!

> <Callout emoji="💡">You can also type your repo without GitHub URL (e.g. “codesandbox/test-sandbox”) to import the repo</Callout>
> 
> The first time when you import a repo, it can take a while. During first-time setup, we clone the repository into the VM and set up the folders for the project. However, after this setup we will create a memory snapshot of the VM, which will ensure that next VM loads (& clones!) will be fast.

Reviewer: We should wrap the following paragraph in the call-out as well. 
```suggestion
<Callout emoji="💡">You can also type your repo without GitHub URL (e.g. “codesandbox/test-sandbox”) to import the repo.
When you first import a repo, it might take a while. This is because, during the initial setup, we clone the repository into the VM and set up the folders for the project. However, after this setup, we will create a memory snapshot of the VM, ensuring that the subsequent VM loads (& clones!) will be fast.
</Callout>
```

Author: I do agree, but I believe the `Callout` count is already pretty high, and Callouts don't support multiple paragraphs right now. That's why I like the idea of having some text in-between to make the article breathe a bit more? I'll update the text a bit to make it less Callout-y, maybe that makes it a bit better.
